qa/comparitor.py

```diff
@@ -0,0 +1,180 @@
+        max_date_string = max(timeseries1_max_date, timeseries2_max_date, actual1_max_date, actual2_max_date)
+
+        min_date = datetime.strptime(min_date_string, "%Y-%m-%d")
+        max_date = datetime.strptime(max_date_string, "%Y-%m-%d")
+
+        delta = max_date - min_date
+
+        dates = []
+        for i in range(delta.days + 1): 
+            day = min_date + timedelta(days=i)
+            dates.append(day.strftime("%Y-%m-%d"))
+        
+        return dates
+    
+    def compareMetrics(self):
+        for date in self.getDates(): 
+            for metric in METRICS: 
+                actual_name = metric.actual_path[-1] if metric.actual_path else None
+                projection_name = metric.projection_path[-1] if metric.projection_path else None
+                
+                projected_value1 = self.getMetricProjectedValue(metric, date, self.snapshot1)
+                projected_value2 = self.getMetricProjectedValue(metric, date, self.snapshot2)
+
+                actual_value1 = self.getMetricActualValue(metric, date, self.snapshot1)
+                actual_value2 = self.getMetricActualValue(metric, date, self.snapshot2)
+
+                if (projected_value1 and projected_value2): 
+                    if metric.isAboveThreshold(projected_value1, projected_value2): 
+                        self._results[projection_name] = MetricDiff(
+                            self.state, 
+                            self.county, 
+                            self.fips, 
+                            date, 
+                            projection_name, 
+                            self.snapshot1, 
+                            self.snapshot2, 
+                            projected_value1, 
+                            projected_value2, 
+                            metric.diff(actual_value1, actual_value2), 
+                            metric.threshold)
+
+                if (actual_value1 and actual_value2): 
+                    if metric.isAboveThreshold(actual_value1, actual_value2): 
+                        self._results[actual_name] = MetricDiff(
+                            self.state, 
+                            self.county, 
+                            self.fips, 
+                            date, 
+                            actual_name, 
+                            self.snapshot1, 
+                            self.snapshot2, 
+                            actual_value1, 
+                            actual_value2, 
+                            metric.diff(actual_value1, actual_value2), 
+                            metric.threshold)
+        return self.results
+            
+    @property
+    def results(self): 
+        return self._results.values()
+
+    def write_to_csv(self, file_name):
+        with open(file_name, 'w+') as csv_file: 
+            write = csv.writer(csv_file) 
+            write.writerows(self.results)  
```

Comment:
We should write the header as well, I'd recommend using the `csv.DictWriter`: https://docs.python.org/3/library/csv.html#csv.DictWriter

Reply:
yesss was just messing around